spec/an_metadata_update_helper_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'tmpdir'
+require_relative './spec_helper'
+
+describe Fastlane::Helper::StandardMetadataBlock do
+  it 'strips any trailing newline when generating the block for a single-line input' do
+    Dir.mktmpdir do |dir|
+      input = "Single line message with new line\n"
+
+      # Generate the input file to convert to .pot block
+      input_path = File.join(dir, 'input')
+      File.write(input_path, input)
+      # Ensure the input has only one line
+      expect(File.read(input_path).lines.count).to eq 1
+
+      # Write the .pot block in an output file
+      output_path = File.join(dir, 'output')
+      File.open(output_path, 'w') do |file|
+        described_class.new('any-key', input_path).generate_block(file)
+      end
```

Comment:
💄 nit: I think we couldn't use a StringIO instead here to avoid having to write to disk there.
(`StringIO` is a ruby built-in class that is quite useful to mock `IO` operations, as it's a `String` that acts like and has the same methods as an `IO` (e.g. a `File`), but do it all in-memory and not on-disk. We should try to use it more often in our specs imho)

I think something like this should work then (not tested):

```suggestion
      output_io = StringIO.new
      described_class.new('any-key', input_path).generate_block(output_io)
```

Then use `expect(output_io.string).to eq …` below.

Reply:
:TIL: ✨ 

That's neat, thanks. Addressed in 28dba6d.